### DISCUSSION TITLE
Apply styling in tailwind.config.js based on light/dark mode

### DISCUSSION THREAD

**Discussion by kkyr** [tailwindlabs/tailwindcss#10932] [Help] [answered]
I'm using Tailwind CSS [typography](https://tailwindcss.com/docs/typography-plugin#adding-custom-color-themes) plugin in order to automatically style HTML elements. I'm customising the code block according the customisation guide on the wiki, which involves defining your own styling inside tailwind.config.js.

This is a sample that I'm using:

```
module.exports = {
  darkMode: 'class',
  theme: {
      typography: {
        DEFAULT: {
          css: {
            'code': {
              'background-color': 'black',
            },
          },
        },
      },
  },
  plugins: [
    require('@tailwindcss/typography'),
  ],
}
```

This makes the code block render with a black background. I would like, however, to render a black background only when light mode is enabled and a white background when dark mode is enabled. I'm familiar with the `dark-` modifier which can be used throughout the HTML filched, but I'm not sure how I can use the modifier inside `tailwind.config.js`.

I've tried adding a new modifier dark under typography (i.e. a sibling to DEFAULT with key "dark") with dark-specific stylings, but this did not change anything.

Note that I need to make this change inside `tailwind.config.js`, the HTML is generated at compile time by a markdown generator so I cannot add css classes directly to the code block.

**Comment by wongjn** [comment, marked as answer]
The object value within the `css` key is parsed as CSS by `postcss-js`, so for dark styles, you can write a separate rule for dark mode:
```js
module.exports = {
  darkMode: 'class',
  theme: {
    typography: {
      DEFAULT: {
        css: {
          'code': {
            'background-color': 'black',
          },
          '.dark code': {
            backgroundColor: 'pink',
          },
        },
      },
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
  ],
};
```

Alternatively, you could use an [element modifier](https://tailwindcss.com/docs/typography-plugin#element-modifiers) if you are able to add classes to the element with the `prose` class:

```html
<div class="prose prose-code:bg-black dark:prose-code:bg-pink-500">
```

**Reply by kkyr** [reply]
Works like a charm, thanks!
